test/docfx.Test/template/XrefTagParserTest.cs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Text.RegularExpressions;
+using Xunit;
+
+namespace Microsoft.Docs.Build
+{
+    public class XrefTagParserTest
+    {
+        [Theory]
+        [InlineData("<xref uid='{{uid}}'/>",
+            "{{#uid}}" +
+            "  {{#href}}" +
+            "    <a href=\"{{href}}\"> {{uid}} </a>" +
+            "  {{/href}}" +
+            "  {{^href}}" +
+            "    <span> {{.}} </span>" +
+            "  {{/href}}" +
+            "{{/uid}}"
+            )]
+        [InlineData("<xref uid='{{namespace}}'/>",
+            "{{#namespace}}" +
+            "  {{#href}}" +
+            "    <a href='{{href}}'> {{uid}} </a>" +
+            "  {{/href}}" +
+            "  {{^href}}" +
+            "    <span> {{.}} </span>" +
+            "  {{/href}}" +
+            "{{/namespace}}"
+            )]
+        [InlineData("<xref uid='{{ uid }}' template='partials/dotnet/xref-name.tmpl' />",
+            "{{#uid}}" +
+            "  {{#href}}" +
+            "    {{> partials/dotnet/xref-name.tmpl}}" +
+            "  {{/href}}" +
+            "  {{^href}}" +
+            "    <span> {{.}} </span>" +
+            "  {{/href}}" +
+            "{{/uid}}"
+            )]
+        [InlineData("<xref uid='{{ . }}'/>",
+            "{{#href}}" +
+            "  <a href=\"{{href}}\"> {{uid}} </a>" +
+            "{{/href}}" +
+            "{{^href}}" +
+            "  <span> {{.}} </span>" +
+            "{{/href}}"
+            )]
+        [InlineData("<xref uid='{{ . }}' template='partials/module/xref-unit-link.tmpl' />",
+            "{{#href}}" +
+            "  {{> partials/module/xref-unit-link.tmpl}}" +
+            "{{/href}}" +
+            "{{^href}}" +
+            "  <span> {{.}} </span>" +
+            "{{/href}}"
+            )]
+        public void ProcessXrefTag(string template, string expected)
+        {
+            template = template.Replace('\'', '"');
+            Assert.Equal(
+                Regex.Replace(expected.Replace('\'', '"'), @"(?<=\}) +| +(?=\{)", ""),
```

Comment:
is there a simpler way of doing the validation without regex? like striping out all newlines and spaces. Guessing the meaning of `(?<=\}) +| +(?=\{)` is no fun :(